Nodejs/Product/Npm/PackageJsonFactory.cs

```diff
@@ -1,14 +1,109 @@
-// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
+using System.Globalization;
+using System.IO;
+using System.Threading;
 using Microsoft.NodejsTools.Npm.SPI;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 
 namespace Microsoft.NodejsTools.Npm
 {
-    public class PackageJsonFactory
+    public static class PackageJsonFactory
     {
-        public static IPackageJson Create(IPackageJsonSource source)
+        public static IPackageJson Create(string fullPathToFile)
         {
-            return source.Package == null ? null : new PackageJson(source.Package);
+            if (string.IsNullOrEmpty(fullPathToFile) && !fullPathToFile.EndsWith("package.json"))
```

Comment:
Should this check be case-insensitive?

Reply:
Also, I would find a null check clearer than `IsNullOrEmpty`.

Reply:
Yes, that should be case-insensitive